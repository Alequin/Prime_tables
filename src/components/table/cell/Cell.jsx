import React from "react"

class Cell extends React.Component{

  constructor(props){
    super(props)
  }

  render(){
    return (
      <section className="cell-container">

      </section>
    )
  }
}

export default Cell
