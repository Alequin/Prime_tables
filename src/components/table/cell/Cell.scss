.cell-container{
  text-align: center;
  width: 50px;
  padding: 5px;
  font-size: 18px;
  border: 1px solid black;
}
