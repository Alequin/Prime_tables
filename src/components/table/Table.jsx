import React from "react"
import PropTypes from 'prop-types'
import Cell from "./cell/Cell.jsx"

class Table extends React.Component{

  constructor(props){
    super(props)
    this.state = {
      row: 0,
      column: 0,
    }
  }

  buildRow(tableBuilder, rowIndex){
    const row = []
    let length = tableBuilder.maxPosition+this.state.column
    if(length > this.maxColumns) length = this.maxColumns+this.state.column

    for(let column=this.state.column; column<length; column++){
      const cellValue = tableBuilder.valueAt(rowIndex, column)
      const info = tableBuilder.infoAt(rowIndex, column)
      row.push(<Cell key={cellValue} value={cellValue} info={info}/>)
    }
    return <div key={rowIndex} className="row">{row}</div>
  }

  buildTable(){
    const table = []
    let length = this.props.tableBuilder.maxPosition+this.state.row
    if(length > this.maxRows) length = this.maxRows+this.state.row
    for(let row=this.state.row; row<length; row++){
      table.push(this.buildRow(this.props.tableBuilder, row))
    }
    return table
  }

  render(){
    return (
      <section className="table-container">
        {this.buildTable()}
      </section>
    )
  }
}

Table.propTypes = {
  tableBuilder: PropTypes.object,
}

export default Table
