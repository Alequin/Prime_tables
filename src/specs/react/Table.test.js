import React from 'react'
import Table from './../../components/table/Table.jsx'
import Cell from './../../components/table/cell/Cell.jsx'
import renderer from 'react-test-renderer'
import assert from "assert"


describe("Table component", function(){

  beforeEach(() => {

  })

  it('renders correctly', () => {
    const tree = renderer
      .create(<Table/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("can generate table", () => {
    let table = new Table({primeNumberCount: 2})
    let expected = [
        <div className="Row">
          <Cell value={" "}/>
          <Cell value={2}/>
          <Cell value={3}/>
        </div>,

        <div className="Row">
          <Cell value={2}/>
          <Cell value={4}/>
          <Cell value={6}/>
        </div>,
        
        <div className="Row">
          <Cell value={3}/>
          <Cell value={6}/>
          <Cell value={9}/>
        </div>
    ]
    let result = table.buildTable()
    assert.deepEqual(result, expected)
  })
})
