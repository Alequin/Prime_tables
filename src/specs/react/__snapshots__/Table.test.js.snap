// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders correctly 1`] = `
<section
  className="table-container"
>
  <div
    className="row"
  >
    <div
      className="cell-container"
      onClick={[Function]}
    >
      _
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      3
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="cell-container"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      6
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="cell-container"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="cell-container"
      onClick={[Function]}
    >
      9
    </div>
  </div>
</section>
`;
